core/src/main/scala/org/apache/spark/config/ConfigBuilder.scala

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.config
```

Comment:
> private[spark] has no impact on java

At what level? Yes, the method in the java class file is public, but it doesn't seem to actually show up in any documentation. e.g. in http://spark.apache.org/docs/latest/api/java/index.html you don't see things like `SparkConf.validateSettings`, which is `private[spark]`.

I don't like putting things in specially-named packages in hope that people will not look at them, but if it means I can get this change moving, I'll move classes into whatever package.